### PR TITLE
Rosetta: Record files (blocks) retrieval

### PR DESCRIPTION
**Pull request by failfmi**
**Detailed description**:
* Skip record files before the first account balance file (cache them).
* Update how latest and by hash blocks are found.
* Update all sql statements to be in `raw` format.
* Add unit tests for `persistence/block`.
* Add unit tests for `services/base`.
* Restructure mocks folder.

**Which issue(s) this PR fixes**:
Fixes #None

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by xin-hedera — CHANGES_REQUESTED** [review]
will review the test cases later

File: hedera-mirror-rosetta/app/persistence/block/block.go [hunk 1]
```diff
@@ -176,3 +227,27 @@ func (br *BlockRepository) constructBlockResponse(rf *recordFile, blockIndex int
...
+
+	var accountBalancesCount int64
+
+	br.dbClient.Raw(selectAccountBalancesCount).Count(&accountBalancesCount)
+	if accountBalancesCount == 0 {
+		return nil, errors.Errors[errors.NodeIsStarting]
+	}
+
+	var startingIndex int64
+	br.dbClient.Raw(selectSkippedRecordFilesCount).Count(&startingIndex)
```
[the remark this answers is not in the document]
Comment (failfmi) [reply hedera-mirror-rosetta/app/persistence/block/block.go:244]: It actually works. (Example from docs -> ```db.Table("deleted_users").Select("count(distinct(name))").Count(&count)
// SELECT count(distinct(name)) FROM deleted_users```

[the remark this answers is not in the document]
Comment (xin-hedera) [reply hedera-mirror-rosetta/app/persistence/block/block.go:250]: thanks for the clarification!

**Review by xin-hedera — DISMISSED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**